invenio_records_rest/utils.py

```diff
@@ -139,7 +166,7 @@ def data(self):
                 response = make_response(jsonify(data), data['status'])
                 response.headers['Location'] = location
                 abort(response)
-            except BuildError:
+            except (BuildError, ValueError):
```

Comment:
There is still though the possibility that some client code specify a `pid_type` that is not defined. In this case `ValueError` would still be triggered.